### DISCUSSION TITLE
Issues running setting everything up natively

### DISCUSSION THREAD

**Discussion by CRhodes-123** [bitcraze/discussions#538] [Q&A]
Hello everyone, 
My name is Collin, and I am a student doing research with crazyflies at the University of Houston. Currently I am setting everything up to run natively on a laptop that uses Ubuntu 20.04.4 LTS, which is the exact same operating system that the Virtual Machine on my laptop uses. I have downloaded every folder from github that is present in the virtual machine, and set the folders to be identical in both name and file paths. I have downloaded visual studio code and downloaded every extension present in the virtual machine. 

When I go to run the update_all.sh program in the terminal, I get this error:
![Update All Script Error](https://user-images.githubusercontent.com/103008098/215230253-d4923cf3-c286-4aba-af7d-3ed8f97dd5a7.png)

I do not know what this error means or how to fix it, so any assistance would be greatly appreciated!

Thanks,
Collin R.

**Comment by ataffanel** [comment]
I am not sure what is happening and I am not sure what you mean by 'downloaded every project from github'. Did you `git clone` all projects or did you download the zip? The `update_all_projects.sh` script assumes the projects are cloned with Git.

From the message it seems that maybe pip is not installed. This could be solved with `sudo apt install python3-pip`.

However, we did not design this script to run anywhere else than in the VM. The main point of the script is to bring all the projects up to date when someone starts the VM for the first time. So what I could suggest is to manually clone and use the project you are interested on working with.

**Reply by CRhodes-123** [reply]
I downloaded the zip files present in the home/bitcraze/Desktop/projects file directory and I also found the files present in the home/bitcraze/Desktop folder and downloaded those too. I unzipped every folder and made the names to be identical. I will try to install pip and see if I can figure out how to clone and get back to you.
From what I understand the files should all be up to date, I just want to be able to run the script later on when files start to need to be updated, and from what I understand it's a relatively easy way to verify that every file that I might need is present

**Reply by ataffanel** [reply]
Zip files might not carry unix metadata for the files and it might cause a problem for git. If you want to reproduce the VM environment you can look at the provisioning script we use to make the VM. For example running these lines will clone all the projects: https://github.com/bitcraze/bitcraze-vm/blob/6b414af820aca57c485f204b82edef53a5c58762/provision.sh#L37-L49 and this is the debian packages that are installed https://github.com/bitcraze/bitcraze-vm/blob/6b414af820aca57c485f204b82edef53a5c58762/provision.sh#L10-L11.

The full script would not work on your machine since it assumes it runs for a "bitcraze" user, but you can pick from it all the information you need to reproduce the VM environment.
